### ISSUE TITLE
After reboot randomly opens some conky's, not all

### ISSUE THREAD

**Issue by Lowrida** [zcot/conky-manager2#31]
I have about 35 conky windows and after a reboot conkymanager does not start all the conky's! It randomly starts some of them, not all. Very annoying! It does not happen all the time, but it always happen after i started a liveusb session and then boot back to my normal linux! Then some (random) of the conky's are started, some not... i have no clue why and were to look for this problem??

**Comment by zcot** [comment]
Interesting. I wonder if that would say that the same thing would happen on any sort of reboot? I'm taking a guess that from using a liveusb you are using reboot? warm restart, as opposed to cold restart. I don't know if that would be relevant but taking a guess that it might.

The conky-manager2, if you are using the autostart setting, and timer, writes a startup shell file location in ~/.conky(that's the default location and based on your settings path) and that would include a timed delay as well as any list of configs to be started.

If you are not instructing conky-manager2 to handle all of the configs, and for example you have your own scripts that might start any of the other configs, then there could likely be a conflict of when some or others would start, and the conky-manager2 script is created to include a killall command, which would kill any other configs that have potentially already started at that point.

You can make sure to set the manager to handle all of them, or go edit the start script and remove the killall line, if you think this might be part of the problem.

Or maybe there is some issue once you reach such a threshold of 35 configs.

The conky-manager2 does not start or run when a re/boot happens, only the created script is run, by the system, based on startup script activity. So my guess would be that there might be a problem with the script, or that regardless of the manager the system itself gets into a weird position of handling the 35 configs all firing up one right after the other.

If it's never happening from a cold boot, then it means there is some delay or otherwise viable condition during cold boot that offers the opportunity for all the scripts to run successfully without any issues.

**Comment by Lowrida** [comment]
I don't think it is because of the threshold of 35 conky's. I recently switched from mint 19.x to mint 20 and before i had no problem of any kind with this many.

When i want to start the liveusb session i sometimes use the "reboot" command from the prompt, but mostly i restart from the menu. Don't know if that would make a difference, i have to try that.

Conkymanager handles all conky's scripts! Start and stop.
I run a few bash scripts, they are started via aplicationstarter. They interact with some of the conky's.

When i enter the desktop it takes about 10 seconds (delay of conky) before the conky's show. If they start and would get killed, then i would notice them before they get terminated, but they do not appear at all.
Next time, before i log on to the desktop, i will view the startup script of conkymanager and check if all the scripts are in there. What if they are not??

When does conkymanager writes to that script? At reboot or immediatly when i check or uncheck a script?

**Comment by Lowrida** [comment]
I can't put my finger on it... if i reboot via startmenu or type reboot in terminal, does not matter. Randomly starts conky's.
I restarted the script, so it kills all conky's that are started and starts them again.. and then randomly other conky's are started (or not)... still the same problem that not all are started!
That is when i just rebooted from a live iso!

When i reboot again and go immediatly to my normal linux all conky's are started!

**Comment by zcot** [comment]
Make the delay 15 seconds, or 20 seconds and see if that solves the problem, because there are many other things that start when you boot the system. You can tailor the timing to make sure the bash scripts are started before the conky configs.

conky-manager2 will re/write the start script, every time you close the program, if you are using the startup option.

**Comment by Lowrida** [comment]
The bash scripts have a failsafe in them. They go into a waiting loop until the conky for that script is loaded and then pops out of that loop and does it's work until the conky goes offline, then it shoots in the waiting loop again. But the conky's do not need these scripts to run..
I use the autostart application app to start the startup script of conkymanager, not the manager itself.
I have the delay increased from 10 to 20 seconds. But i find it strange that coming from a i5 4-core 3,2ghz without problems, now on a i5 6-core/12 threads 4,9ghz with 32gb ram and problems starting a few scripts....... that makes no sense to me?

**Comment by Lowrida** [comment]
Well... the 20 seconds seems to work. I have rebooted numerous times today and they all get started.
Still strange that a newer system with much more compute power, more ram, new os, fresh os with less installed, has trouble with starting these scripts..

**Comment by zcot** [comment]
Good to hear.

On a systemd system things can be started much more synchronously than a straight line queue of individual tasks. Doing that on a high performance system can mean that within the first seconds there is less predictability during any given second as to what is what or where, at least until the starts have finished.

Aside from the system itself being fully stabilized, you already know that when you start a conky config that it takes 2 or 3 passes to become stable as far as all of the pieces of it being fully functional, depending on what variable or function you might be involved with. So if there is a config trying to use for example `gw_iface`, and there are more, then the result can be unpredictable if run too quickly on a system that is not quite done yet being fully finished with all boot processes.

-must be something involving that stuff.

**Comment by Lowrida** [comment]
If that were so, then it would always be the same conky script that fails to start, but it is random!
But for 2 weeks now i had no problem at all with the 20 seconds delay.. so i guess to many programs loading at the same time..
